test/index.test.js

```diff
@@ -1,12 +1,146 @@
 /*global describe, it */
 
 var expect = require("chai").expect,
-	db = require("../index");
+	getConnectionOptions = require("./getConnectionOptions"),
+	Database = require("../index");
 
 describe("iw-mysql-wrapper", function () {
 
-	it("should be defined", function () {
-		expect(db).to.be.a("function");
+	describe("Database.paramify()", function () {
+
+		it("should return empty values/tokens when list is empty", function () {
+			var result = Database.paramify([], "pfx");
+			expect(result).to.eql({
+				values: {},
+				tokens: []
+			});
+		});
+
+		it("should create a values/tokens map for the list", function () {
+			var result = Database.paramify(["one", "two"], "pfx");
+			expect(result).to.eql({
+				values: {
+					"pfx0": "one",
+					"pfx1": "two"
+				},
+				tokens: [":pfx0", ":pfx1"]
+			});
+		});
+
+	});
+
+	describe("Database()", function () {
+
+		var db;
+		beforeEach(function () {
+			db = new Database(getConnectionOptions());
+		});
+
+		afterEach(function (done) {
+			db.end(done);
+		});
+
+		describe("paramify()", function () {
+
+			it("should return empty values/tokens when list is empty", function () {
+				var result = db.paramify([], "pfx");
+				expect(result).to.eql({
+					values: {},
+					tokens: []
+				});
+			});
+
+			it("should create a values/tokens map for the list", function () {
+				var result = db.paramify(["one", "two"], "pfx");
+				expect(result).to.eql({
+					values: {
+						"pfx0": "one",
+						"pfx1": "two"
+					},
+					tokens: [":pfx0", ":pfx1"]
+				});
+			});
+
+		});
+
+		describe("queryFormat()", function () {
+
+			it("should replace tokens with escaped values", function () {
+				var query = "SELECT * FROM test WHERE col1 = :value1 and col2 = :value2 and col3 = :value1;";
+				var values = {
+					value1: "ohai",
+					value2: "ohai'; bobby tables"
+				};
+				expect(db.queryFormat(query, values)).to.eql("SELECT * FROM test WHERE col1 = 'ohai' and col2 = 'ohai\\'; bobby tables' and col3 = 'ohai';");
+			});
+
+			it("should leave alone values which aren't present", function () {
```

Comment:
You do get an error, if you leave out a :param - it comes from mysql, because the SQL is invalid. The `formatQuery` function is copy/pasted from `mysql` documentation - I'm really not sure it's worth adding/changing any behaviour, but hey, look - you know how to write code yourself if you think any features need to be added!